src/touch.js

```diff
@@ -0,0 +1,89 @@
+var common = require('./common');
+var fs = require('fs');
+
+module.exports = _touch;
+
+//@
+//@ ### touch(file)
+//@
+//@ Examples:
+//@
+//@ ```javascript
+//@ touch('file.txt');
+//@ ```
+//@
+//@ Creates a file if it does not exist. Sets the file's mtime and ctime to NOW if the file
+//@ does exist
+function _touch(opts, file) {
+	opts = common.parseOptions(opts, {
+		'a': 'atime',
+		'c': 'no_create',
+		'd': 'date',
+		'm': 'mtime',
+		'r': 'reference',
+	});
+
+	if (!file) {
+		common.error('no paths given');
+	}
+
+	if (typeof file !== 'string') {
+		common.error('file arg should be a single file path');
+	}
+
+	// if the string ends with "/" then stop
+	if (file.indexOf('/', file.length - 1) !== -1) {
+		common.error('cannot touch directory ' + file);
```

Comment:
I ran the following and saw no errors:

```
$ mkdir foo/; touch foo/
```

Could we change this to just skip the file instead? That seems more in-line with POSIX behavior.